app/src/main/java/com/arygm/quickfix/MainActivity.kt

```diff
@@ -119,29 +121,33 @@ fun QuickFixApp(testBitmapPP: Bitmap?, testLocation: Location = Location()) {
 
   val preferencesViewModel: PreferencesViewModel =
       viewModel(factory = PreferencesViewModel.Factory(LocalContext.current.dataStore))
+  val userPreferencesViewModel: PreferencesViewModelUserProfile =
+      viewModel(factory = PreferencesViewModelUserProfile.Factory(LocalContext.current.dataStore))
+
+    val workerViewModel: ProfileViewModel =
+        viewModel(key = "workerViewModel", factory = ProfileViewModel.WorkerFactory)
+    val categoryViewModel: CategoryViewModel = viewModel(factory = CategoryViewModel.Factory)
+    val locationViewModel: LocationViewModel = viewModel(factory = LocationViewModel.Factory)
 
   var isOffline by remember { mutableStateOf(!isConnectedToInternet(context)) }
 
-  var currentAppMode by remember { mutableStateOf<String?>(null) }
+  var currentAppMode by remember { mutableStateOf(AppMode.USER) }
   Log.d("userContent", "Current App Mode is empty: $currentAppMode")
   LaunchedEffect(Unit) {
     Log.d("userContent", "Loading App Mode")
     currentAppMode =
         when (loadAppMode(preferencesViewModel)) {
-          "User" -> AppContentRoute.USER_MODE
-          "Worker" -> AppContentRoute.WORKER_MODE
+          "User" -> AppMode.USER
+          "Worker" -> AppMode.WORKER
           else -> {
-            AppContentRoute.USER_MODE
+            AppMode.WORKER
```

Comment:
It would be nice if the default mode was USER in all cases, but that's okay, it should never happen.